scorep/user.py

```diff
@@ -117,12 +116,23 @@ def __exit__(self, exc_type, exc_value, traceback):
         if (callable(self.func)
             and instrumenter.get_instrumenter().get_registered()
                 and not self.user_region_name):
-            """
-            looks like there is a decorator, we are registered and the name is not specified by the user,
-            so we do not need to do anything. The Instrumentation will take care.
-            """
+            # * we are a decorator
```

Comment:
Can we make this whole exit function mirror the enter function? It took me a while to review all this and using the same structure as the enter function might be helpful at least for that.
Should also the instrumenter be disabled like it is done in enter?